pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -745,12 +745,6 @@ func (npm *nominatedPodMap) add(p *v1.Pod, nodeName string) {
 		}
 	}
 	npm.nominatedPodToNode[p.UID] = nnn
-	for _, np := range npm.nominatedPods[nnn] {
```

Comment:
The value of `nnn` seems not be same as the value of [nominated node](https://github.com/kubernetes/kubernetes/blob/180af4240c42906f072112e7ac7818f4b3574c5a/pkg/scheduler/internal/queue/scheduling_queue.go#L758)?

Reply:
If pod exist in `npm.nominatedPods[nnn]`, it should be deleted when `npm.delete(p)` invoked?

Reply:
There is a possibility that pod was nominated to a different node other than the `nodeName` passed to `add` function here.

you can see the map is set here, 
```
npm.nominatedPodToNode[p.UID] = nnn
```

while the `nnn` is get from the existing `nominatedPodToNode`
```
nnn, ok := npm.nominatedPodToNode[p.UID]
```

Reply:
> There is a possibility that pod was nominated to a different node other than the `nodeName` passed to `add` function here.

Emm...seems like the npm.delete(p) queries the "nominatedPodToNode" first, "nominatedPodToNode" means there is only one nominated node for the pod. so that should be ok even the pass different nodeName to add function here.

Reply:
+1, seems like it is good even they are different.

Reply:
The `delete` is deleting the nominated pod of `npm.nominatedPodToNode[p.UID]`,  the check of existence is actually on the node passed in, is there a status one pod is nominated to different nodes?